### PR TITLE
Create Fibonacci Implementation

### PR DESCRIPTION
**Pull request by zamonier**

### CODE REVIEW

File: recursion/src/test/java/FibonacciRecursiveImplementationTest.java [hunk 1]
```diff
@@ -13,16 +13,16 @@
 
     private static Stream<Arguments> test_getFibonacciNumber() {
         return Stream.of(
-                of(0, 0),
-                of(1, 1),
+                of(1, 0),
```
Comment (PaulFrmBrn) [inline recursion/src/test/java/FibonacciRecursiveImplementationTest.java]: Are you sure about that?
https://en.wikipedia.org/wiki/Fibonacci_number

File: recursion/src/main/java/FibonacciImpl.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+public class FibonacciImpl implements Fibonacci {
+
+  @Override
+  public int at(int index) {
+    if (index < 1) {
+      throw new IllegalArgumentException();
```
Comment (PaulFrmBrn) [inline recursion/src/main/java/FibonacciImpl.java]: Don't you want to provide message?

Comment (zamonier) [reply recursion/src/main/java/FibonacciImpl.java]: Based on lecture

Comment (zamonier) [reply recursion/src/main/java/FibonacciImpl.java]: fixed

File: recursion/src/test/java/FibonacciRecursiveImplementationTest.java (hunk 1 above)
Comment (PaulFrmBrn) [reply recursion/src/test/java/FibonacciRecursiveImplementationTest.java]: ping

Comment (zamonier) [reply recursion/src/test/java/FibonacciRecursiveImplementationTest.java]: fixed

**Pull request merged**